controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1594,43 +1594,46 @@ def test_expand_volume_with_no_space_in_pool_exception(self):
 
 class TestIdentityServer(BaseControllerSetUp):
 
-    @patch.object(CSIControllerServicer, "get_identity_config")
+    @patch("controller.controller_server.common_config.config.identity")
     def test_identity_plugin_get_info_succeeds(self, identity_config):
         plugin_name = "plugin-name"
         version = "1.1.0"
-        identity_config.side_effect = [plugin_name, version]
+        identity_config.name = plugin_name
+        identity_config.version = version
         request = Mock()
         context = Mock()
         request.volume_capabilities = []
         response = self.servicer.GetPluginInfo(request, context)
         self.assertEqual(response, csi_pb2.GetPluginInfoResponse(name=plugin_name, vendor_version=version))
 
-    @patch.object(CSIControllerServicer, "get_identity_config")
+    @patch("controller.controller_server.common_config.config.identity")
     def test_identity_plugin_get_info_fails_when_attributes_from_config_are_missing(self, identity_config):
         request = Mock()
         context = Mock()
 
-        identity_config.side_effect = ["name", Exception(), Exception(), "1.1.0"]
-
+        identity_config.mock_add_spec(spec=["name"])
         response = self.servicer.GetPluginInfo(request, context)
         context.set_code.assert_called_once_with(grpc.StatusCode.INTERNAL)
         self.assertEqual(response, csi_pb2.GetPluginInfoResponse())
 
+        identity_config.mock_add_spec(spec=["version"])
         response = self.servicer.GetPluginInfo(request, context)
         self.assertEqual(response, csi_pb2.GetPluginInfoResponse())
         context.set_code.assert_called_with(grpc.StatusCode.INTERNAL)
 
-    @patch.object(CSIControllerServicer, "get_identity_config")
+    @patch("controller.controller_server.common_config.config.identity")
     def test_identity_plugin_get_info_fails_when_name_or_value_are_empty(self, identity_config):

```

Comment:
```suggestion
    def test_identity_plugin_get_info_fails_when_name_or_version_are_empty(self, identity_config):

```
while you're at it...